packages/lesmis/src/api/mutations/useUpdateFavouriteDashboardItem.js

```diff
@@ -2,37 +2,75 @@
  * Tupaia
  *  Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
  */
-import { useMutation } from 'react-query';
-
+import { useMutation, useQueryClient } from 'react-query';
 import { post } from '../api';
 
-const useFavouriteDashboardItem = () =>
-  useMutation(dashboardItemCode =>
-    post('userFavouriteDashboardItems', {
-      data: {
-        state: 'favourite',
-        dashboardItemCode,
+// Instead of refetching dashboard for 'isFavourite' config update, we update the current existing dashboard data.
```

Comment:
What's the rationale for not re-fetching the data here? It feels a little heavy handed / unsafe to manually update the state here?

Reply:
I thought re-fetch has to call the re-fetch function from dashboard `fetchQuery` callback function, which will be heavy to pass that to here. 

But I just found that we can refetch a specific query from `queryClient`, that's super handy!